int32 num_obj_label
float32[] pcaCentroid
